subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -815,6 +815,128 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
         !file("build/classes/java/main/ignored/io/ignored/IgnoredExample.class").exists()
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/20394")
+    @Requires(TestPrecondition.JDK16_OR_LATER)
```

Comment:
Aren't records a JDK14 feature?

Reply:
They are a JDK 14 _preview_ feature, but only stabilized in 16 [[source]](https://openjdk.java.net/projects/jdk/16/).

Reply:
same here